new-log-viewer/src/services/decoders/JsonlDecoder.ts

```diff
@@ -0,0 +1,240 @@
+import dayjs from "dayjs";
+
+import {JsonlDecodeOptionsType} from "../../typings/decoders";
+import {
+    INVALID_TIMESTAMP_VALUE,
+    LOG_VERBOSITY,
+} from "../../typings/logs";
+import {
+    Decoder,
+    DecodeResultType,
+} from "./Decoder";
+
+
+/**
+ * Type of values in JSON structures.
+ * Reference: https://www.json.org/json-en.html
+ */
+type JsonValue = string |
+    number |
+    boolean |
+    { [key: string]: JsonValue | null } |
+    Array<JsonValue | null>;
+
+/**
+ * Type of JSON object structure, which is a collection of name/value pairs.
+ * Reference: https://www.json.org/json-en.html
+ */
+type JsonObject = { [key: string]: JsonValue | null };
+
+class JsonlDecoder implements Decoder {
+    static TEXT_DECODER = new TextDecoder();
+
+    readonly #dataArray: Uint8Array;
+
+    #logEvents: JsonObject[] = [];
+
+    #textPattern: string =
+        "%d{yyyy-MM-dd HH:mm:ss.SSS} [%process.thread.name] %log.level %message%n";
+
+    #datePattern: string = "%d{yyyy-MM-dd HH:mm:ss.SSS}";
+
+    #dateFormat: string = "YYYY-MM-DD HH:mm:ss.SSS ZZ";
+
+    #timestampPropName: string = "ts";
+
+    #verbosityPropName: string = "level";
+
+    #propertyNames: string[] = [];
+
+    static #convertLogbackDateFormatToDayjs (dateFormat: string) {
+        // Fix year
+        dateFormat = dateFormat.replace("yyyy", "YYYY");
+        dateFormat = dateFormat.replace("yy", "YY");
+
+        // Fix day
+        dateFormat = dateFormat.replace("dd", "D");
+        dateFormat = dateFormat.replace("d", "D");
+
+        return dateFormat;
+    }
+
+    #extractDateFormat () {
+        // Extract date format from text string
+        const dateFormatMatch = this.#textPattern.match(/%d\{(.+?)}/);
+        if (null === dateFormatMatch) {
+            console.warn(
+                "Unable to find date format string in #textPattern:",
+                this.#textPattern
```

Comment:
```suggestion
    /**
    * Sets the text pattern for the decoder and extracts the date format and property names from it.
    *
    * @param {string} pattern - The text pattern to set for the decoder.
    * @return {void}
    */
    #setTextPattern (pattern: string) {
```

Reply:
For functions returning `void` (doesn't return a value), we can omit the `@return` in the docs.